File: njob-rest/src/main/java/com/github/nsorin/njobrest/model/Recruiter.java
package com.github.nsorin.njobrest.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.validation.constraints.Email;
import javax.validation.constraints.Pattern;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.github.nsorin.njobrest.model.validation.ValidationPattern;

import lombok.Getter;
import lombok.Setter;

@Entity
public class Recruiter {

    @GeneratedValue
    @Id
    @Getter
    @Setter
    private Integer id;

    @Getter
    @Setter
    @Pattern(regexp = ValidationPattern.PERSON_NAME)
    private String firstName;

    @Getter
    @Setter
    @Pattern(regexp = ValidationPattern.PERSON_NAME)
    private String lastName;

    @Getter
    @Setter
    @Email
    private String email;

    @Getter
    @Setter
    @Pattern(regexp = ValidationPattern.PHONE_NUMBER)
    private String phone;

    @JsonBackReference
    @ManyToOne
    @Getter
    @Setter
    private Company company;

}

File: njob-frontend/src/store/models/Recruiter.js
import { Model } from "@vuex-orm/core"
import Company from "./Company"

export default class Recruiter extends Model {

    static entity = 'recruiter'

    static fields() {
        return {
            id: this.number(null),
            firstName: this.string(''),
            lastName: this.string(''),
            email: this.string(''),
            phone: this.string(''),
            company_id: this.number(null),
            company: this.belongsTo(Company, 'company_id')
        }
    }
}